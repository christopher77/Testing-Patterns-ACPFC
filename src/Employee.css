.tree li:only-child {
  padding-top: 0;
}

.tree li::before,
.tree li::after {
  content: "";
  position: absolute;
  top: 0;
  right: 50%;
  border-top: 1px solid #ccc;
  width: 50%;
  height: 1em;
}

.tree li::before,
.tree li::after {
  content: "";
  position: absolute;
  top: 0;
  right: 50%;
  border-top: 1px solid #ccc;
  width: 50%;
  height: 1em;
}

.tree li::after {
  right: auto;
  left: 50%;
  border-left: 1px solid #ccc;
}

.tree li:first-child::after {
  border-radius: 5px 0 0 0;
}

.tree li:last-child::before {
  border-right: 1px solid #ccc;
  border-radius: 0 5px 0 0;
}

.tree li:first-child::before,
.tree li:last-child::after {
  border: 0 none;
}

.tree ul ul::before {
  content: "";
  position: absolute;
  top: 0;
  left: 50%;
  border-left: 1px solid #ccc;
  width: 0;
  height: 1em;
}
