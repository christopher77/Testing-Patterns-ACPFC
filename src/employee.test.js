import React from "react";
import { render } from "@testing-library/react";
import Employee from "./employee";

test("Render employee", () => {
  const employee = [
    {
      id: "Fri Jun 07 2019 17:08:11 GMT-0500 (Peru Standard Time)",
      name: "CEO",
      title: "CEO",
      parentId: null
    },
    {
      id: "Fri Jun 07 2019 17:08:16 GMT-0500 (Peru Standard Time)",
      name: "Carlos",
      title: "Dev",
      parentId: "Fri Jun 07 2019 17:08:11 GMT-0500 (Peru Standard Time)"
    },
    {
      id: "Fri Jun 07 2019 17:08:21 GMT-0500 (Peru Standard Time)",
      name: "Daniel",
      title: "Dev",
      parentId: "Fri Jun 07 2019 17:08:16 GMT-0500 (Peru Standard Time)"
    }
  ];

  function getChildren(id) {
    return employee.filter(employee => employee.parentId === id);
  }
  const handleCreateChild = jest.fn();
  const { asFragment } = render(
    <Employee
      employee={employee}
      getChildren={getChildren}
      handleCreateChild={handleCreateChild}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
