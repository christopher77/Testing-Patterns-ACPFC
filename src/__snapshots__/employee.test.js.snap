// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render employee 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  text-align: center;
  list-style-type: none;
  padding: 1em 0.5em 0 0.5em;
}

.emotion-0 {
  border: 1px solid #ccc;
  padding: 0.5em 0.75em;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  border-radius: 5px;
  color: #333;
  position: relative;
  top: 1px;
}

<li
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
       - 
      <form>
        <input
          aria-label="Employee name"
          id="employee-name"
          placeholder="Employee name"
          type="text"
          value=""
        />
      </form>
    </div>
  </li>
</DocumentFragment>
`;
